### PR TITLE
Views specs and n+1 problems

### PR DESCRIPTION
**Pull request by Marcraphael12**
In this PR we are writing integration tests with capybara, and we are solving N+1 problems with eager loading.

#### n+1
- Make sure that the N+1 problem is solved when fetching all posts and their comments for a user.
    - Do not use the bullet gem in this exercise - instead analyze the logs as showed in this [article](https://dev.to/junko911/rails-n-1-queries-and-eager-loading-10eh)
    - **In your PR description include the screenshots of your app's logs before and after the fix**
    - before: 
before reduce1
   - after: 
after reduce1

#### Integration specs
- Use Capybara to write integration tests for each view in your project.
- Think on the flow of things and group the tests of page in a way that makes sense. For example, one test would check for the things in the page and another test for clicking links on it.

- User index page:
  - I can see the username of all other users.
  - I can see the profile picture for each user.
  - I can see the number of posts each user has written.
  - When I click on a user, I am redirected to that user's show page.

- user show page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see the user's bio.
  - I can see the user's first 3 posts.
  - I can see a button that lets me view all of a user's posts.
  - When I click a user's post, it redirects me to that post's show page.
  - When I click to see all posts, it redirects me to the user's post's index page.

- User post index page:
  - I can see the user's profile picture.
  - I can see the user's username.
  - I can see the number of posts the user has written.
  - I can see a post's title.
  - I can see some of the post's body.
  - I can see the first comments on a post.
  - I can see how many comments a post has.
  - I can see how many likes a post has.
  - I can see a section for pagination if there are more posts than fit on the view.
  - When I click on a post, it redirects me to that post's show page.

- Post show page:
  - I can see the post's title.
  - I can see who wrote the post.
  - I can see how many comments it has.
  - I can see how many likes it has.
  - I can see the post body.
  - I can see the username of each commentor.
  - I can see the comment each commentor left.

### CODE REVIEW

File: spec/views/users/index.html.erb_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    @post = Post.create(author: @user, title: 'Awakening', text: 'I choose the red pill')
+    @post_two = Post.create(author: @user, title: 'Training', text: 'I know kung fu')
+    @post_three = Post.create(author: @user, title: 'The Oracle', text: 'There is no spoon')
+    @post_four = Post.create(author: @user, title: 'Choice', text: 'I believe I can save Morpheus')
+    @post_five = Post.create(author: @user, title: 'Agent encounter', text: 'still only human')
+    @comment = Comment.create(post_id: @post, author_id: @user, text: 'time to fight')
+    @comment_two = Comment.create(post_id: @post_two, author_id: @user, text: 'time to fight')
+    visit user_posts_path(@user)
+  end
+
```
Comment (1GORDON) [inline spec/views/users/index.html.erb_spec.rb:8]: Great job team 🤝
- As stated in the project requirements, kindly include tests that redirect to the user's show page when you click on a user.

File: spec/views/users/show.html.erb_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    @first_post.update(comments_counter: @first_post.comments.count)
+    visit user_posts_path(@first_post.author, @first_post)
+  end
+
+  scenario 'displays the post title' do
+    visit user_posts_path(@first_post.author, @first_post)
+    expect(page).to have_content(@first_post.author.name)
+  end
+
+  # ################################################################*************************************
```
Comment (1GORDON) [inline spec/views/users/show.html.erb_spec.rb]: Great job team🤝
- As stated in the project requirements, kindly ensure your tests cover all the listed in the image below.
user-show

File: spec/views/users/index.html.erb_spec.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
+require 'rails_helper'
+
```
Comment (1GORDON) [inline spec/views/users/index.html.erb_spec.rb:2]: Great job team 🤝
- Some of your tests are still failing the test checks. Kindly ensure all your tests are passing. failing-tests

**Review by 1GORDON — CHANGES_REQUESTED** [review]
##
## Hi @Marcraphael12, and @tomasesquivelgc ✋

Good job so far!

## Highlights!
- Has Capybara gem in `Gemfile` for integration test ✔️ 
- Has featured tests for `User` **index page** ✔️
- Has featured tests for `User` show page ✔️
- Has featured tests for `User` **post index page** ✔️
- Has featured tests for `User` **post show page** ✔️

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend considering them as they can improve your code._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@1GORDON` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/users/index.html.erb_spec.rb (hunk 3 above)
Comment (Marcraphael12) [reply spec/views/users/index.html.erb_spec.rb:2]: image

**Review by OyePriscilla — APPROVED** [review]
Hi @Marcraphael12 and @tomasesquivelgc 👏 👏 👏 👏, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights
Implemented tests for posts' and users' page ✔️ 
All test passed ✔️ 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**